webpack/assets/javascripts/react_app/routes/ExternalRoutes.js

```diff
@@ -0,0 +1,22 @@
+import { Route } from 'react-router-dom';
+import React from 'react';
+import { addGlobalFill } from '../components/common/Fill/GlobalFill';
+import { updateLocation, removeRailsContent } from './RoutingService';
+
+export const addGlobalRoutes = (id, routes) =>
```

Comment:
`routes` is an array of all of the routes of a plugin, wouldn't it better than adding each route as an argument? especially when the mechanism atm to represents routes  is by array

Reply:
It doesn't mean you should add each item manually, you can always use it as:
```jsx
const routes = ['some', 'items'];
const globalRoutes = addGlobalRoutes(id, ...routes);
// Or
const route = 'some-item';
const globalRoutes = addGlobalRoutes(id, route);
```

Another benefit, the `routes` will always have a default value of an empty array

Reply:
But this one
```js
const route1 = 'some-item1';
const route2 = 'some-item1';
const globalRoutes = addGlobalRoutes(id, route1, route2);
```
won't work, but the function signature `addGlobalRoutes(id, ...routes);` suggests that it can be valid

Reply:
It should work as well